keras_cv/layers/preprocessing/random_flip.py

```diff
@@ -0,0 +1,201 @@
+        seed=None,
+        bounding_box_format=None,
+        **kwargs
+    ):
+        super().__init__(seed=seed, force_generator=True, **kwargs)
+        self.mode = mode
+        self.seed = seed
+        if mode == HORIZONTAL:
+            self.horizontal = True
+            self.vertical = False
+        elif mode == VERTICAL:
+            self.horizontal = False
+            self.vertical = True
+        elif mode == HORIZONTAL_AND_VERTICAL:
+            self.horizontal = True
+            self.vertical = True
+        else:
+            raise ValueError(
+                "RandomFlip layer {name} received an unknown mode="
+                "{arg}".format(name=self.name, arg=mode)
+            )
+        self.auto_vectorize = True
+        self.bounding_box_format = bounding_box_format
+
+    def augment_label(self, label, transformation, **kwargs):
+        return label
+
+    def augment_image(self, image, transformation, **kwargs):
+        flipped_output = image
+
+        def flip_horizontal():
+            return tf.image.flip_left_right(flipped_output)
+
+        flipped_output = tf.cond(
+            tf.cast(transformation["flip_horizontal"], dtype=tf.bool),
+            flip_horizontal,
+            lambda: flipped_output,
+        )
+
+        def flip_vertical():
+            return tf.image.flip_up_down(flipped_output)
+
+        flipped_output = tf.cond(
+            tf.cast(transformation["flip_vertical"], dtype=tf.bool),
+            flip_vertical,
+            lambda: flipped_output,
+        )
+        flipped_output.set_shape(image.shape)
+        return flipped_output
+
+    def get_random_transformation(self, **kwargs):
+        flip_horizontal = False
+        flip_vertical = False
+        if self.horizontal:
+            flip_horizontal = self._random_generator.random_uniform(shape=[]) > 0.5
+        if self.vertical:
+            flip_vertical = self._random_generator.random_uniform(shape=[]) > 0.5
+        return {
+            "flip_horizontal": flip_horizontal,
+            "flip_vertical": flip_vertical,
+        }
+
+    def augment_bounding_boxes(
+        self, bounding_boxes, transformation=None, image=None, **kwargs
+    ):
+        if self.bounding_box_format is None:
+            raise ValueError(
+                "`RandomFlip()` was called with bounding boxes,"
+                "but no `bounding_box_format` was specified in the constructor."
+                "Please specify a bounding box format in the constructor. i.e."
+                "`RandomFlip(bounding_box_format='xyxy')`"
+            )
+
+        bounding_boxes = bounding_box.convert_format(
+            bounding_boxes,
+            source=self.bounding_box_format,
+            target="xyxy",
+            images=image,
+        )
+        image_shape = tf.shape(image)
+        h = tf.cast(image_shape[H_AXIS], dtype="float32")
+        w = tf.cast(image_shape[W_AXIS], dtype="float32")
+        bounding_boxes_out = tf.identity(bounding_boxes)
```

Comment:
is the identity call necessary here?  Does something ail without it?

Reply:
removed